addons/binding/org.openhab.binding.silvercrestwifisocket/src/main/java/org/openhab/binding/silvercrestwifisocket/handler/SilvercrestWifiSocketHandler.java

```diff
@@ -0,0 +1,351 @@
+
+import static org.openhab.binding.silvercrestwifisocket.SilvercrestWifiSocketBindingConstants.WIFI_SOCKET_CHANNEL_ID;
+
+import java.math.BigDecimal;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.silvercrestwifisocket.SilvercrestWifiSocketBindingConstants;
+import org.openhab.binding.silvercrestwifisocket.entities.SilvercrestWifiSocketRequest;
+import org.openhab.binding.silvercrestwifisocket.entities.SilvercrestWifiSocketResponse;
+import org.openhab.binding.silvercrestwifisocket.enums.SilvercrestWifiSocketRequestType;
+import org.openhab.binding.silvercrestwifisocket.exceptions.MacAddressNotValidException;
+import org.openhab.binding.silvercrestwifisocket.utils.NetworkUtils;
+import org.openhab.binding.silvercrestwifisocket.utils.ValidationUtils;
+import org.openhab.binding.silvercrestwifisocket.utils.WifiSocketPacketConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SilvercrestWifiSocketHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jaime Vaz - Initial contribution
+ */
+public class SilvercrestWifiSocketHandler extends BaseThingHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SilvercrestWifiSocketHandler.class);
+
+    private String hostAddress;
+    private String macAddress;
+    private Long updateInterval = SilvercrestWifiSocketBindingConstants.DEFAULT_REFRESH_INTERVAL;
+
+    private final WifiSocketPacketConverter converter = new WifiSocketPacketConverter();
+    private ScheduledFuture<?> keepAliveJob;
+    private long latestUpdate = -1;
+
+    /**
+     * Default constructor.
+     *
+     * @param thing the thing of the handler.
+     * @throws MacAddressNotValidException if the mac address isn't valid.
+     */
+    public SilvercrestWifiSocketHandler(final Thing thing) throws MacAddressNotValidException {
+        super(thing);
+        this.saveMacAddressFromConfiguration(this.getConfig());
+        this.saveHostAddressFromConfiguration(this.getConfig());
+        this.saveUpdateIntervalFromConfiguration(this.getConfig());
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        if (channelUID.getId().equals(WIFI_SOCKET_CHANNEL_ID)) {
```

Comment:
It can potentially always been called. If a REFRESH command is received, the handler should try to send a state update for the according channel on the next possible occasion.